### PR TITLE
[WIP] Univariate intbase

### PR DESCRIPTION
**Pull request by srajangarg**
A simple implementation of `UnivariateIntBase`. Have started to interface the current implementation of `UnivariateIntPolynomial` to `UnivariateInt`. Still have to port all methods like `add_poly` etc.

Later, `UnivariateIntPolynomial` will be removed. Also, will try and merge more and more methods to the base class whenever possible.

### CODE REVIEW

**Comment by srajangarg** [comment]
@isuruf Any help on the vector initialization, when using Piranha as `integer_class`?

File: symengine/univariate_int.h [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+namespace SymEngine
+{
+
+template <typename Coeff, typename Container, typename Poly>
+class UnivariateIntBase : public Basic
+{
+protected:
+    unsigned int degree_;
+    RCP<const Symbol> var_;
+    Container container_;
```
Comment (isuruf) [inline symengine/univariate_int.h]: What's going to be the container for other types?

**Comment by isuruf** [comment]
What's the error and what did you try?

File: symengine/univariate_int.h (hunk 1 above)
Comment (srajangarg) [inline symengine/univariate_int.h]: Probably, `flint::fmpq_polyxx` and `piranha::polynomial<integer_class, piranha::monomial<unsigned int>>`. I will make changes to the overall structure if something doesn't fit well.

**Comment by srajangarg** [comment]
You can see it on travis. It's basically that a vector of piranha integers can't be initialized in the way given. But, that particular way is the most intuitive, don't you think?

File: symengine/univariate_int.h (hunk 1 above)
Comment (isuruf) [inline symengine/univariate_int.h]: Then `map_int_Expr` is not the right fit here. Maybe something like `UnivariateExprPolynomial` would do. (Not the implementation we have right now, but the one @irislq has started on)

Comment (srajangarg) [inline symengine/univariate_int.h]: But why? What would be the reason behind that?

Comment (isuruf) [inline symengine/univariate_int.h]: Yes, I mixed up `map_uint_mpz` and `map_int_Expr`. The point of a templated function is to write common code. There's nothing common about `map_uint_mpz` and `piranha::polynomial`. How can they be used to write code that is common for both?

**Comment by srajangarg** [comment]
@isuruf Putting this on hold for some time

**Pull request closed without merging**